fix(number-field): fix step=any to support integer and decimal number (#485)
Refinitiv/refinitiv-ui: documents/src/pages/elements/number-field.md

```

## Set input step

> The step attribute specifies the interval between valid numbers. For instance, when `step="2"`, valid values Would only be even numbers e.g. 2,4,6,8... Alternatively, specify the step `any`. This means any value is allowed.

Reviewer: ```suggestion
The step attribute specifies the interval between valid numbers. For instance, when `step="2"`, valid values Would only be even numbers e.g. 2,4,6,8... Alternatively, specify the `step="any"` to allow any value. 
```
